doc/CMakeLists.txt

```diff
@@ -182,6 +182,9 @@ if(${DOXYGEN_FOUND})
         list(APPEND DOXYGEN_TARGET_DEPENDENCIES "${html_images_binary_dir}")
 
     endforeach()
+    # Unset loop variables to avoid bugs later on.
+    unset(html_binary_dir)
```

Comment:
OK. The comment and unset are a little, er ... unsettling though. We don't usually do that at the end of a variables useful lifetime so it makes me wonder why it's being done here. I'm fine with it like this if you think it should be there though.